test/test_cost.py

```diff
@@ -2357,8 +2357,10 @@ def test_td3_notensordict(self, observation_key, reward_key, done_key):
             loss_val_td = loss(td)
             torch.manual_seed(0)
             loss_val = loss(**kwargs)
-            for i, key in enumerate(loss_val_td.keys()):
-                torch.testing.assert_close(loss_val_td.get(key), loss_val[i])
+            for i in loss_val:
+                assert i in loss_val_td.values(), f"{i} not in {loss_val_td.values()}"
+            # for i, key in enumerate(loss_val_td.keys()):
+            # torch.testing.assert_close(loss_val_td.get(key), loss_val[i])
```

Comment:
This is dangerous as keys in the tensordict get ordered by name but output tuple loss_val doesn't. So for now im just checking if all values in the loss_val tuple are also in the loss_val_td.


```
# actor metadata
metadata = {
            "state_action_value_actor": state_action_value_actor.mean().detach(),
        }
# value metadata
        metadata = {
            "td_error": td_error,
            "next_state_value": next_target_qvalue.mean().detach(),
            "pred_value": current_qvalue.mean().detach(),
            "target_value": target_value.mean().detach(),
        }
# out tensordict
        td_out = TensorDict(
            source={
                "loss_actor": loss_actor,
                "loss_qvalue": loss_qval,
                **metadata_actor,
                **metadata_value,
            },
            batch_size=[],
        )
```
loss_vals will be in that order `(loss_actor, loss_qvalue, state_action_value_actor, next_state_value, pred_value, target_value)`
However, as the items are getting ordered in the TD by the keys the output tensordict has actually this order:
 `(loss_actor, loss_qvalue, next_state_value, pred_value, state_action_value_actor, target_value)`

Reply:
dispatch returns the keys in the order of out_keys.
So this is predictable, we can just do
```python
            for i, key in enumerate(loss.out_keys):
                torch.testing.assert_close(loss_val_td.get(key), loss_val[i])
```
does that solve the problem?